plugin-testing-framework/script/testBuilder.ts

```diff
@@ -0,0 +1,210 @@
+
+            if (this.scenarioPath) {
+                before(() => {
+                    return projectManager.setupScenario(TestConfig.testRunDirectory, TestConfig.TestNamespace, TestConfig.templatePath, this.scenarioPath, targetPlatform);
+                });
+            }
+            
+            this.testBuilders.forEach(testBuilder => {
+                if (!(this.hasOnly && !testBuilder.only)) testBuilder.create(coreTestsOnly, projectManager, targetPlatform);
+            });
+        });
+    }
+}
+
+/** A test case to be run by the Plugin Testing Framework. */
+export type TestCase = (projectManager: ProjectManager, targetPlatform: platform.IPlatform, done: MochaDone) => void;
+
+/** Use this class to create a test through mocha.it */
+export class TestBuilderIt extends TestBuilderBase {
+    /** The name of the test */
+    private testName: string;
+    /** The test to be run */
+    private test: TestCase;
+    /** Whether or not the test should be run when "--core" is supplied */
+    private isCoreTest: boolean;
+    
+    /**
+     * testName - used as the expectation in the call to it
+     * test - the test to provide to it
+     * isCoreTest - whether or not the test should run when "--core" is supplied
+     * only - if true, use it.only
+     */
+    constructor(testName: string, test: TestCase, isCoreTest: boolean, options?: { only?: boolean, skip?: boolean }) {
+        super(options);
+        
+        this.testName = testName;
+        this.test = test;
+        this.isCoreTest = isCoreTest;
+    }
+    
+    /**
+     * Called to create the test suite by the initializeTests function
+     * 
+     * coreTestsOnly - Whether or not only core tests are to be run
+     * projectManager - The projectManager instance that these tests are being run with
+     * targetPlatform - The platform that these tests are going to be run on
+     */
+    public create(coreTestsOnly: boolean, projectManager: ProjectManager, targetPlatform: platform.IPlatform): void {
+        if (!this.skip && (!coreTestsOnly || this.isCoreTest)) {
+            it(this.testName, this.test.bind(this, projectManager, targetPlatform));
+        }
+    }
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////
+// Mocha mimicry
+
+export interface ITestBuilderContextDefintion {
+    (description: string, tests: TestBuilderBase[], scenarioPath?: string): TestBuilderBase;
+    only(description: string, tests: TestBuilderBase[], scenarioPath?: string): TestBuilderBase;
+    skip(description: string, tests: TestBuilderBase[], scenarioPath?: string): TestBuilderBase;
+}
+
+/**
+ * Returns a hybrid type that mimics mocha's describe object.
+ */
+function getDescribe(): ITestBuilderContextDefintion {
+    var describer = <ITestBuilderContextDefintion>function(description: string, tests: TestBuilderBase[], scenarioPath?: string): TestBuilderBase {
+        return new TestBuilderDescribe(description, tests, scenarioPath);
```

Comment:
Instead of returning a new instance of the "TestBuilderDescribe" or "TestBuilderIt" to the user, do you think you can push the new instance to a global array so that the user doesn't have to deal with keeping it in an array himself?